### PR TITLE
[Bugfix] Enforce cross-feature dependency checker in pipeline

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

In PR #15692, we introduced a cross-feature dependency checker in the pipeline. However, there were issues preventing the checker from failing the pipeline and blocking the merge when dependencies were detected. In this PR, we resolve those issues, ensuring that the pipeline fails if any cross-feature dependencies are identified.

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
In PR #15692, we introduced a cross-feature dependency checker in the pipeline. However, there were issues preventing the checker from failing the pipeline and blocking the merge when dependencies were detected. In this PR, we resolve those issues, ensuring that the pipeline fails if any cross-feature dependencies are identified.

#### How did you do it?
In this PR, we resolve those issues, ensuring that the pipeline fails if any cross-feature dependencies are identified.

#### How did you verify/test it?
Tested by pipeline itself. If there is a dependency, the pipeline will fail.

### CODE REVIEW

File: .azure-pipelines/dependency-check.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ steps:
 
     pip3 install natsort
 
-    CHECK_RESULT=$(python3 ./.azure-pipelines/dependency_check/dependency_check.py tests)
-    if [[ "$CHECK_RESULT" == "True" ]]; then
+    python3 ./.azure-pipelines/dependency_check/dependency_check.py tests
+    if [[ $? -eq 1 ]]; then
```
Comment (wangxin) [inline .azure-pipelines/dependency-check.yml:8]: Why only rc=1 is considered as failure? What about other rc values that may be added in the future?
Is this format more robust and extensible? `if [[ $? -neq 0 ]]; then`?

Comment (yutongzhang-microsoft) [reply .azure-pipelines/dependency-check.yml]: modified

File: .azure-pipelines/dependency-check.yml [hunk 2]
```diff
@@ -4,8 +4,8 @@ steps:
 
     pip3 install natsort
 
-    CHECK_RESULT=$(python3 ./.azure-pipelines/dependency_check/dependency_check.py tests)
-    if [[ "$CHECK_RESULT" == "True" ]]; then
+    python3 ./.azure-pipelines/dependency_check/dependency_check.py tests
+    if [[ $? -ne 0 ]]; then
       echo "##vso[task.complete result=Failed;]Condition check failed."
```
Comment (wangxin) [inline .azure-pipelines/dependency-check.yml:9]: Why set variable `result` here?

Comment (yutongzhang-microsoft) [reply .azure-pipelines/dependency-check.yml:9]: As I searched, ##vso[task.complete result=Failed;] is used to explicitly mark a task as failed. And exit 1 is a double confirm to fail the pipeline.

**Review by wangxin — APPROVED** [review]

**Pull request merged**